### PR TITLE
feat(edgeless): copilot selection widget

### PR DESCRIPTION
**Pull request by doouding**
Press ctrl or command (mac) when dragging to activate AI.

### CODE REVIEW

**Review by doodlewind — COMMENTED** [review]
I think we should land all edgeless AI features in different PRs, can we make drag selection into a feature flag and land this PR earlier?

**Comment by doouding** [comment]
I was thinking of using stacked PR. This PR would be the base of other Edgeless AI related PRs and it will get merged once all its sub-PRs is ready. Which on do you prefer?

**Review by doodlewind — COMMENTED** [review]
https://github.com/toeverything/blocksuite/assets/7312949/779435ab-989a-4cfa-ad37-a152deab2cc9

After releasing the ai selection box, it will unexpectedly select an element on the canvas.

File: packages/blocks/src/root-block/widgets/edgeless-ai-selection/index.ts [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  on,
+  requestConnectedFrame,
+} from '../../../_common/utils/event.js';
+import type { AIToolController } from '../../edgeless/controllers/tools/ai-tool.js';
+import type { EdgelessRootBlockComponent } from '../../index.js';
+
+export const AFFINE_EDGELESS_AI_WIDGET = 'affine-edgeless-ai-widget';
+
+@customElement('affine-edgeless-ai-widget')
+export class EdgeelssAIWidget extends WidgetElement<EdgelessRootBlockComponent> {
```
Comment (doodlewind) [inline packages/blocks/src/root-block/widgets/edgeless-ai-selection/index.ts:17]: typo

File: packages/blocks/src/_common/types.ts [hunk 2]
```diff
@@ -239,7 +243,8 @@ export type EdgelessTool =
   | ConnectorTool
   | EraserTool
   | FrameTool
-  | FrameNavigatorTool;
+  | FrameNavigatorTool
+  | AITool;
```
Comment (doodlewind) [inline packages/blocks/src/_common/types.ts:247]: `CopilotSelectionTool`

**Review by doodlewind — APPROVED** [review]
Thanks for the feat! Should land this after some naming updates.

File: packages/blocks/src/root-block/edgeless/services/tools-manager.ts [hunk 3]
```diff
@@ -290,33 +313,53 @@ export class EdgelessToolsManager {
...
 
-      return this.currentController.onContainerPointerDown(e);
+    if (
+      isMiddleButtonPressed(pointEvt) ||
+      isRightButtonPressed(pointEvt) ||
+      metaKeyPressed
+    ) {
+      const isRightButton = isRightButtonPressed(pointEvt);
+      const targetTool = (
+        isRightButton || metaKeyPressed
```
Comment (congzhou09) [inline packages/blocks/src/root-block/edgeless/services/tools-manager.ts:326]: @doouding 
Here may should be "isRightButton && metaKeyPressed" if you meant rightbutton-dragging with ctrl or command (mac) pressed.

Comment (doouding) [reply packages/blocks/src/root-block/edgeless/services/tools-manager.ts:326]: Actuallly not, here I mean right button dragging OR meta key pressed.

Comment (congzhou09) [reply packages/blocks/src/root-block/edgeless/services/tools-manager.ts:326]: > Actuallly not, here I mean right button dragging OR meta key pressed.

Oh I thought right button dragging could pan too. I take back.

**Pull request merged**